cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1284,7 +1346,11 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 
 	// TODO: do we actually need fsck if we are mounting a mapper device?
 	// probably not?
-	if err := doSystemdMount(unlockRes.Device, boot.InitramfsDataDir, fsckSystemdOpts); err != nil {
+	devPath := unlockRes.Device
+	if unlockRes.IsDecryptedDevice {
+		devPath = unlockRes.DecryptedDevice
```

Comment:
this can always use this no, without the if ?

Reply:
as discussed, this can now always use FsDevice if `err == nil`